File: 3566-find-the-sequence-of-strings-appeared-on-the-screen/find-the-sequence-of-strings-appeared-on-the-screen.go
func stringSequence(t string) []string {
    a := make([]string, 0)
    s := "a"
    a = append(a, s)
    i := 0

    for s != t {
        if s[i] == t[i] {
            i++
            s += "a"
            a = append(a, s)
        } else {
            for s[i] != t[i] {
                s = s[:i] + string(s[i]+1) + s[i+1:] 
                a = append(a, s)
            }
        }
    }
    return a
}

File: 3566-find-the-sequence-of-strings-appeared-on-the-screen/find-the-sequence-of-strings-appeared-on-the-screen.cpp
class Solution {
public:
    vector<string> stringSequence(string t) {
        vector<string> a;
        string s = "a";
        a.push_back(s);
        
        int i = 0;
        
        while (s != t) {
            if (s[i] == t[i]) {
                i++;
                s += 'a';
                a.push_back(s);
            } else {
                while (s[i] != t[i]) {
                    s[i]++;
                    a.push_back(s);
                }
            }
        }
        
        return a;
    }
};
